### ISSUE TITLE
Design Explorations: Error States

### ISSUE THREAD

**Issue by melchoyce** [WordPress/block-library#11]
We need to figure out all the potential errors and where there can be points of failure:
- Block can’t install (your internet craps out, etc.)
- What happens if you don’t have permissions
- What else?

Related: #8

**Comment by melchoyce** [comment]
**If we can't preview the block for some reason, like a WordPress.org outage:**

![image](https://user-images.githubusercontent.com/2846578/61001121-f83ac380-a32c-11e9-9335-3990b6564d39.png)

@ck3lee do you know if we have existing error strings for something like this? If not, we'll want to write one.

**If the installation fails (but the block can preview):**

![image](https://user-images.githubusercontent.com/2846578/61001177-1dc7cd00-a32d-11e9-8d7d-0bda59dc7b93.png)

---

Is there any other point in this flow that might produce an error?

**Comment by melchoyce** [comment]
**Permissions**

If you don't have administrative permissions (or any custom permission that allows installation of new plugins), let's show the existing message for search results:

<img src="https://user-images.githubusercontent.com/2846578/61001715-28369680-a32e-11e9-9b73-d19e9e634839.png" width="400" />

**Comment by boemedia** [comment]
In the last case, is it possible to show the blocks but greyed out, with a message saying: 
"Your current role doesn't allow installing new blocks. Contact your website's administrator."

**Comment by melchoyce** [comment]
Maybe like:

![image](https://user-images.githubusercontent.com/2846578/61081155-8f1e8300-a3f4-11e9-8256-a1cc6bb14f54.png)

**Comment by karmatosed** [comment]
I like that as an option over having greyed out. I feel if we showed greyed out blocks it could taunt the user.

**Comment by ck-lee** [comment]
> do you know if we have existing error strings for something like this?

No, I couldn't find any existing error strings. I only found a couple that are generic and technical like
`The response is not a valid JSON response.`. So, we might need to write them.

**Comment by ck-lee** [comment]
LGTM! I think the two scenarios: "can't preview the block" and "installation fails" should capture all the unexpected failure points. I will drop a comment here if I can think of anything else.

**Comment by ck-lee** [comment]
@melchoyce, I'm finding it very tricky to implement the actions in the notice bar. I am finding out more from the Gutenberg dev team now. 

In case, it is a mission to add the actions like "Retry" or "Remove block", is it OK to remove them?

**Comment by melchoyce** [comment]
As long as we can add them to a later roadmap and follow up, I think it's okay to drop from the v1.

**Comment by ck-lee** [comment]
Thank you @melchoyce. Looks like we can do a "Retry". 😄 

By default all "non-url actions" in the notice are rendered as buttons instead of links. I think it might be semantically more appropriate for a11y. But happy to re-style that as a link too.

Still a WIP. But here is what is looks like:
![screencast 2019-07-18 10-55-16](https://user-images.githubusercontent.com/5370495/61417862-d654c880-a94c-11e9-8d36-e388ce5585ea.gif)

**Comment by ck-lee** [comment]
And for user without permission to install a block. 

This is how it looks like:

![screencast 2019-07-18 10-57-12](https://user-images.githubusercontent.com/5370495/61418015-42373100-a94d-11e9-8d3e-82e2a49590ab.gif)

**Comment by karmatosed** [comment]
@melchoyce I wonder if the 'no blocks' could be an opportunity for a cute little graphic? Like a sad block. It's just an idea.

**Comment by karmatosed** [comment]
Just checking back on this, could it be in v1 to explore a little visual variation beyond text?

**Comment by melchoyce** [comment]
Should we explore this in Gutenberg core? We're hooking in to the core "no results" pattern.

**Comment by karmatosed** [comment]
Sure, I will make issue for this also.

**Comment by melchoyce** [comment]
Thanks, feel free to assign it to me.

**Comment by karmatosed** [comment]
@melchoyce https://github.com/WordPress/gutenberg/issues/17122 is where started that issue.
